yt_dlp/YoutubeDL.py

```diff
@@ -603,10 +609,31 @@ def __init__(self, params=None, auto_init=True):
         except Exception as e:
             self.write_debug(f'Failed to enable VT mode: {e}')
 
-        self._allow_colors = Namespace(**{
-            type_: not self.params.get('no_color') and supports_terminal_sequences(stream)
-            for type_, stream in self._out_files.items_ if type_ != 'console'
-        })
+        color_policy = self.params.get('color') or {}
+        self.params['color'] = {
+            name: (
+                'nocolor' if self.params.get('no_color') else
+                color_policy if isinstance(color_policy, str) else
+                color_policy.get(name) or 'auto')
+            for name in ('screen', 'stdout', 'stderr')
+        }
+        term_allow_color = os.environ.get('TERM', '').lower() != 'dumb'
+
+        def process_color_policy(name, stream):
+            policy = self.params['color'].get(name)
+            if policy == 'never':
+                return False
+            if policy == 'always':
+                return True
+            if policy == 'nocolor':
+                return 'nocolor'
+            return term_allow_color and supports_terminal_sequences(stream)
+
+        self._allow_colors = Namespace(
+            out=process_color_policy('stdout', self._out_files.out),
+            error=process_color_policy('stderr', self._out_files.error),
+            screen=process_color_policy('screen', self._out_files.screen),
+        )
```

Comment:
How about doing `color_mapping = {sys.stderr: True, sys.stdout: 'nocolor'}` and then `color_mapping.get(stream)`?

Reply:
I don't understand.

Reply:
The idea is you can pass in a dict from stream to color policy. Not sure if this is easier for user though, but certainly easier to implement for us. We translate from `'stderr'` to `sys.stderr` in `__init__` then
```py
YoutubeDL({"color": {sys.stderr: True, sys.stdout: "no_color"}})
```

Reply:
I'm ok with that too.

Reply:
I did it the other way around still, since using `'stdout': False` should be easier, but im not quite satisfied with the implementation. Thoughts on the current code?